test/unit/specs/utils/decodeData.spec.js

```diff
@@ -0,0 +1,22 @@
+import decodeData from '@/utils/decodeData';
+
+describe('decodeData', () => {
+  it('returns empty string for empty string', () => {
+    const data = "";
+
+    expect(decodeData(data)).toEqual("");
+  });
+
+  it('returns empty string for undefined data', () => {
+    const data = undefined;
+    const empty="";
+
+    expect(decodeData(data)).toEqual(""); 
+  });
+
+  it('decodes unicode strings', () => {
+    const data = 's\\xe9'
+
+    expect(decodeData(data)).toBe('s\xe9');
```

Comment:
Did you also test this with strings with unicode characters such as `'string \u1234'`?

Reply:
Added decoding for unicode characters and tests for the same.